nrf52-hal-common/src/spim.rs

```diff
@@ -55,6 +61,92 @@ impl_spim_ext!(
 /// - The over-read character is hardcoded to `0`.
 pub struct Spim<T>(T);
 
+impl<T> Transfer<u8> for Spim<T> where T: SpimExt
+{
+   type Error = Error;
+
+    fn transfer<'w>(&mut self, words: &'w mut [u8]) -> Result<&'w [u8], Error> {
```

Comment:
There is a fair amount of duplicated code here between `Spim<T>::write()` and this method. Would it make sense to refactor some of the common parts into an inline-able function?

Reply:
Another option could be to have the `write` function call the `transfer` function, basically only handling the CS GPIO?

Reply:
I will see if I could make a nice refactoring of it, almost the same code is on three places.